generator/src/main/java/com/scottlogic/deg/generator/fieldspecs/RowSpecDataBagSourceFactory.java

```diff
@@ -0,0 +1,101 @@
+package com.scottlogic.deg.generator.fieldspecs;
+
+import com.scottlogic.deg.generator.Field;
+import com.scottlogic.deg.generator.generation.FieldSpecValueGeneratorFactory;
+import com.scottlogic.deg.generator.generation.GenerationConfig;
+import com.scottlogic.deg.generator.generation.databags.DataBag;
+import com.scottlogic.deg.generator.generation.databags.DataBagSource;
+import com.scottlogic.deg.generator.generation.databags.MultiplexingDataBagSource;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+public class RowSpecDataBagSourceFactory {
+    private final FieldSpecValueGeneratorFactory generatorFactory;
+
+    public RowSpecDataBagSourceFactory(FieldSpecValueGeneratorFactory generatorFactory) {
+        this.generatorFactory = generatorFactory;
+    }
+
+    public DataBagSource createDataBagSource(RowSpec rowSpec){
+        if (rowSpec instanceof ReductiveRowSpec){
```

Comment:
This _is believe_ is a necessary evil within Java to ensure it can be used with this statement:


See `DecisionTreeDataGenerator`
```
rowSpecs.map(dataBagSourceFactory::createDataBagSource)
```

I don't know if Java uses [double-dispatch](https://en.wikipedia.org/wiki/Double_dispatch), therefore I don't know if it would automatically call the 'reductive' overload or would always call this instance.

Reply:
The method is private, but it you think it helps.
Renamed method for clarity.